nncf/common/sparsity/schedulers.py

```diff
@@ -89,6 +89,17 @@ def current_sparsity_level(self) -> float:
         """
         return self._current_level
 
+    def load_state(self, state: Dict[str, Any]) -> None:
+        super().load_state(state)
+        if self._current_level is None:
+            self._current_level = self._calculate_sparsity_level()
+
+    def get_state(self) -> Dict[str, Any]:
+        if self._current_level is None:
+            self._current_level = self._calculate_sparsity_level()
```

Comment:
For which purposes is it needed here?

Reply:
We have a test (not sure if it is a valid one, but probably this case can be present sometimes).
https://github.com/openvinotoolkit/nncf/blob/8bcf5d243e53afca9511457dffbd7f6048669ba9/tests/torch/sparsity/test_common.py#L93-L122
After creating the scheduler (`self._current_level` is still `None`), the parameters of the scheduler are changed manually and  `get_compression_state()` is called.
```
ctrl.scheduler.current_step = 100
ctrl.scheduler.current_epoch = 5
compression_state = ctrl.get_compression_state()
```
To be precise here, the scheduler should recalculate the sparsity level in accordance with the current epoch and step.